app.py

```diff
@@ -82,7 +88,7 @@ def on_encode_clicked(self, button):
         buffer = self.message_to_encode.get_buffer()
         (start, stop) = buffer.get_bounds()
         message_to_encode = buffer.get_text(start, stop, 0)
-        encoded_message = encode(message_to_encode, 7)
+        encoded_message = encode(message_to_encode, self.shift_input.get_value_as_int())
```

Comment:
This will get the value of the spinbutton as int so we can pass it directly to the encode module